WcaOnRails/app/controllers/users_controller.rb

```diff
@@ -13,11 +13,16 @@ def index
     respond_to do |format|
       format.html { }
       format.json do
-        @users = User
-        if params[:search]
-          @users = @users.where("name LIKE :input OR wca_id LIKE :input OR email LIKE :input", { input: "%#{params[:search]}%" })
+        @users = User.joins("INNER JOIN Countries ON iso2 = country_iso2")
+        params[:search]&.split&.each do |part|
+          like_query = %w(users.name wca_id email Countries.name).map do |column|
+            column + " LIKE :part"
+          end.join(" OR ")
+          @users = @users.where(like_query, { part: "%#{part}%" })
         end
-        if params[:sort]
+        if params[:sort] == "country"
+          @users = @users.order("Countries.name #{params[:order] == "asc" ? "asc" : "desc"}")
```

Comment:
I'm actually okay with mutating `params[:order]`, so I'd probably do something like this:

```ruby
params[:order] = params[:order] == "asc" ? "asc" : "desc"
```

Creating a new variable is fine too. You can put this wherever you think is best. I normally like putting things that are logically connected nearby, but I do also like the pattern of sanity checking at the top of methods.